include/boost/dll/detail/windows/shared_library_impl.hpp

```diff
@@ -71,20 +81,21 @@ class shared_library_impl {
         if (!!(mode & load_mode::append_decorations)) {
             mode &= ~load_mode::append_decorations;
 
-            handle_ = boost::winapi::LoadLibraryExW((sl.native() + L".dll").c_str(), 0, static_cast<native_mode_t>(mode));
-            if (!handle_) {
-                // MinGW loves 'lib' prefix and puts it even on Windows platform
-                const boost::filesystem::path load_path = (sl.has_parent_path() ? sl.parent_path() / L"lib" : L"lib").native() + sl.filename().native() + L".dll";
-                handle_ = boost::winapi::LoadLibraryExW(
-                    load_path.c_str(),
-                    0,
-                    static_cast<native_mode_t>(mode)
-                );
-            }
+            const boost::filesystem::path load_path = decorate(sl);
+            handle_ = boost::winapi::LoadLibraryExW(
+                load_path.c_str(),
+                0,
+                static_cast<native_mode_t>(mode)
+            );
 
             if (handle_) {
                 return;
             }
+            if (boost::filesystem::exists(load_path)) {
+                // decorated path exists : current error is not a bad file descriptor
+                ec = boost::dll::detail::last_error_code();
```

Comment:
That's the wrong way to use last_error_code. The last_error_code function must be called immediately after the system function in wich error code we are interested in. Putting filesystem::exists between the LoadLibrary and last_error_code breaks the error reporting.

So just move the `ec = boost::dll::detail::last_error_code();` line before the `if` statement. That will do the job.